python/lsst/pipe/supertask/super_task.py

```diff
@@ -101,119 +44,72 @@ class SuperTask(Task):
+                            probably be removed)
         :return: The SuperTask Class
         """
 
+        # No spaces allowed in names for SuperTasks
+        # TO DO: need better validation here, some other characters may be disallowed
         if name is None:
-            name = getattr(self, "_default_name", None)
+            name = getattr(self, "_DefaultName", None)
+        if name is not None:
+            name = name.replace(" ", "_")
         super(SuperTask, self).__init__(config, name, parent_task, log)  # Initiate Task
 
-        self._parser = None
-        self.input = Struct()  # input and output will be deprecated
-        self.output = None
-        self._activator = activator
         self._completed = False  # Hook to indicate whether a SuperTask was completed
-        self.list_config = None  # List of configuration items
-        self.name = self.name.replace(" ", "_")  # No spaces allowed in names for SuperTasks
+        self._list_config = None  # List of configuration items
         self._task_kind = 'SuperTask'  # To differentiate between this and Task
 
         self.log.info('%s was initiated' % self.name)  # For debugging only,  to be removed
 
     @property
     def name(self):
         """
-        name of task as property
-        :return: Name of (Super)Task
-        """
-        return self._name
-
-    @name.setter
-    def name(self, name):
-        """
-        Sets the name of task
-        :param name: new name for (Super)Task
-        """
-        self._name = name
-
-    @property
-    def activator(self):
-        """
-        Hook for the name of activator as a property
-        :return: Name of activator invoking this (Super)Task
-        """
-        return self._activator
+        Name of this super-task.
 
-    @activator.setter
-    def activator(self, activator):
+        Note that base class defines `getName()` method which returns the same information,
+        need to think if this property is needed.
         """
-        Sets the name of the activator
-        :param activator:  new name of activator
-        """
-        self._activator = activator
+        return self._name
 
     @property
     def task_kind(self):
         """
-        Defined task_kind as a property
-        :return: Name the kind of task SuperTask or Task
+        (obsolete) String representing kind othis task (e.g. "SuperTask")
         """
         return self._task_kind
 
-    @task_kind.setter
-    def task_kind(self, task_kind):
-        """
-        task_kind setter
-        :param task_kind: kind of task, SuperTask or Task
-        """
-        self._task_kind = task_kind
-
-    @property
-    def parent_name(self):
-        """
-        parent name as property
-        """
-        return self._parent_name
-
     @property
     def completed(self):
         """
-        Return the status of (Super)Task
+        Status of (Super)Task, True if task has completed.
```

Comment:
sounds good, the idea was to have the 'user' explicitly decide whether a task is completed base on some custom criteria (other than the task exit with 0).  We can leave it around and remove it later (instead of the other way around)